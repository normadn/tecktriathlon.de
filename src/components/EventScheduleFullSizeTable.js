import React from "react";
import { css } from "glamor";
import moment from "moment-timezone";

const EventScheduleFullSizeTable = (props) => (
  <table
    {...css({
      borderRadius: "16px",
      fontWeight: "500",
      overflow: "hidden",
      // borderCollapse: "collapse",
      borderSpacing: 0,
      margin: "14px 0",
      width: "100%",
      "> tbody tr td, > tbody tr th": {
        padding: "12px 32px",
        textAlign: "left"
      },
      "> tbody tr:nth-child(2) td": {
        paddingTop: "24px"
      },
      "> tbody tr:not(:first-child) td:first-child": {
        borderLeft: "1px solid #e4e4e4"
      },
      "> tbody tr:not(:first-child) td:last-child": {
        borderRight: "1px solid #e4e4e4"
      },
      "> tbody tr:last-child td": {
        borderBottom: "1px solid #e4e4e4",
        paddingBottom: "24px !important"
      },
      "> tbody tr:last-child td:first-child": {
        borderBottomLeftRadius: "16px"
      },
      "> tbody tr:last-child td:last-child": {
        borderBottomRightRadius: "16px"
      }
    })}
  >
    <tbody>
      <tr
        css={{
          backgroundColor: "#C81C2E",
          color: "white"
        }}
      >
        <th>Time</th>
        <th>Duration</th>
        <th>Event</th>
        <th>Location</th>
      </tr>
      {props.data.map((item) => (
        <tr
          css={{ backgroundColor: "#F6F6F6" }}
          key={`${item.eventName} ${item.startDate}`}
        >
          <td>{moment(item.startDate).format("h:mm\xa0A")}</td>
          <td>{item.duration}</td>
          <td>
            <div>
              {item.eventName}
            </div>
            <div css={{
              color: "#4d4d4d",
              marginTop: "5px",
              // fontStyle: "italic",
              fontSize: "13px"
            }}>
              {item.subtitle}
            </div>
          </td>
          <td>{item.location}</td>
        </tr>
      ))}
    </tbody>
  </table>
);

export default EventScheduleFullSizeTable;
