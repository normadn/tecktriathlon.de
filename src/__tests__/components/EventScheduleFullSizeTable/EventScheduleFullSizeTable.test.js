import React from "react";
import EventScheduleFullSizeTable from "../../../components/EventScheduleFullSizeTable";
import { shallow } from "enzyme";
import moment from "moment-timezone";

const schedule = [
  {
    startDate: "2019-02-01T17:00:00",
    endDate: "2019-02-01T20:00:00",
    eventName: "Busses arrive/registration",
    location: "Kingston Hall"
  },
  {
    startDate: "2019-02-01T17:00:00",
    endDate: "2019-02-01T20:00:00",
    eventName: "Dinner - Presented by iGan Partners (Sandwiches)",
    location: "Kingston Hall"
  },
  {
    startDate: "2019-02-01T20:00:00",
    endDate: "2019-02-01T22:00:00",
    eventName: "Opening Ceremonies",
    location: "Kingston Hall"
  },
  {
    startDate: "2019-02-01T22:00:00",
    endDate: "2019-02-01T22:30:00",
    eventName: "Team Formation",
    location: "BioSci Auditorium"
  },
  {
    startDate: "2019-02-01T22:00:00",
    endDate: "2019-02-03T10:00:00",
    eventName: "Hacking Begins",
    location: ""
  },
  {
    startDate: "2019-02-01T23:00:00",
    endDate: "2019-02-02T00:00:00",
    eventName:
      "Shipping Products Fast: How to Build Fully-Fledged APIs and Services in minutes using StdLib",
    location: "Walter Light Hall - Room 210"
  },
  {
    startDate: "2019-02-02T00:00:00",
    endDate: "2019-02-02T01:00:00",
    eventName: "Midnight Snack (Noodles) - Provided by StdLib",
    location: "Walter Light Hall Atrium"
  },
  {
    startDate: "2019-02-02T00:00:00",
    endDate: "2019-02-02T01:00:00",
    eventName:
      "Adding Image and Voice Intelligence to Your Apps with Microsoft Cognitive Services",
    location: "Goodwin Hall - Room 254"
  },
  {
    startDate: "2019-02-02T00:30:00",
    endDate: "2019-02-02T01:00:00",
    eventName: "Getting Started With Computer Vision Using Clarifai",
    location: "Walter Light Hall - Room 210"
  },
  {
    startDate: "2019-02-02T01:00:00",
    endDate: "2019-02-02T02:00:00",
    eventName: "Hardware Opens",
    location: "ILC Design Bay"
  },
  {
    startDate: "2019-02-02T01:00:00",
    endDate: "2019-02-02T02:00:00",
    eventName: "Slideshow Karaoke",
    location: "Walter Light Hall - Room 210"
  },
  {
    startDate: "2019-02-03T00:00:00",
    endDate: "2019-02-03T01:00:00",
    eventName: "Snack (Bubble Tea)",
    location: "Walter Light Hall Auditorium"
  },
  {
    startDate: "2019-02-03T01:00:00",
    endDate: "2019-02-03T02:00:00",
    eventName: "Escape Room",
    location: "ILC - Room 119"
  },
  {
    startDate: "2019-02-03T07:30:00",
    endDate: "2019-02-03T08:30:00",
    eventName: "Breakfast",
    location: "ILC - Room 109"
  },
  {
    startDate: "2019-02-03T09:00:00",
    endDate: "2019-02-03T10:00:00",
    eventName: "Hacking Ends",
    location: ""
  },
  {
    startDate: "2019-02-03T13:00:00",
    endDate: "2019-02-03T15:00:00",
    eventName: "Closing Ceremonies",
    location: "Kingston Hall"
  }
];

const addDurations = (events) => {
  return events.map((item) => {
    let duration = moment(item.endDate).diff(item.startDate, "minutes", true);
    if (duration > 59) {
      duration = Math.round((duration / 60) * 100) / 100;
      item.duration = `${duration} hr`;
    } else {
      item.duration = `${duration} min`;
    }
    return duration !== 1 ? { ...item, duration: `${item.duration}s` } : item;
  });
};

describe("EventScheduleFullSizeTable", () => {
  it("renders correctly", () => {
    expect(
      shallow(<EventScheduleFullSizeTable data={addDurations(schedule)} />)
    ).toMatchSnapshot();
  });
});
