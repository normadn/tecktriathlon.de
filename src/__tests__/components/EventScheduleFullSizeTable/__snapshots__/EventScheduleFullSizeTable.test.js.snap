// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventScheduleFullSizeTable renders correctly 1`] = `
<table
  data-css-1o8yid1=""
>
  <tbody>
    <tr
      css={
        Object {
          "backgroundColor": "#C81C2E",
          "color": "white",
        }
      }
    >
      <th>
        Time
      </th>
      <th>
        Duration
      </th>
      <th>
        Event
      </th>
      <th>
        Location
      </th>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Busses arrive/registration 2019-02-01T17:00:00"
    >
      <td>
        5:00 PM
      </td>
      <td>
        3 hrs
      </td>
      <td>
        <div>
          Busses arrive/registration
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        Kingston Hall
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Dinner - Presented by iGan Partners (Sandwiches) 2019-02-01T17:00:00"
    >
      <td>
        5:00 PM
      </td>
      <td>
        3 hrs
      </td>
      <td>
        <div>
          Dinner - Presented by iGan Partners (Sandwiches)
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        Kingston Hall
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Opening Ceremonies 2019-02-01T20:00:00"
    >
      <td>
        8:00 PM
      </td>
      <td>
        2 hrs
      </td>
      <td>
        <div>
          Opening Ceremonies
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        Kingston Hall
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Team Formation 2019-02-01T22:00:00"
    >
      <td>
        10:00 PM
      </td>
      <td>
        30 mins
      </td>
      <td>
        <div>
          Team Formation
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        BioSci Auditorium
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Hacking Begins 2019-02-01T22:00:00"
    >
      <td>
        10:00 PM
      </td>
      <td>
        36 hrs
      </td>
      <td>
        <div>
          Hacking Begins
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td />
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Shipping Products Fast: How to Build Fully-Fledged APIs and Services in minutes using StdLib 2019-02-01T23:00:00"
    >
      <td>
        11:00 PM
      </td>
      <td>
        1 hr
      </td>
      <td>
        <div>
          Shipping Products Fast: How to Build Fully-Fledged APIs and Services in minutes using StdLib
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        Walter Light Hall - Room 210
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Midnight Snack (Noodles) - Provided by StdLib 2019-02-02T00:00:00"
    >
      <td>
        12:00 AM
      </td>
      <td>
        1 hr
      </td>
      <td>
        <div>
          Midnight Snack (Noodles) - Provided by StdLib
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        Walter Light Hall Atrium
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Adding Image and Voice Intelligence to Your Apps with Microsoft Cognitive Services 2019-02-02T00:00:00"
    >
      <td>
        12:00 AM
      </td>
      <td>
        1 hr
      </td>
      <td>
        <div>
          Adding Image and Voice Intelligence to Your Apps with Microsoft Cognitive Services
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        Goodwin Hall - Room 254
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Getting Started With Computer Vision Using Clarifai 2019-02-02T00:30:00"
    >
      <td>
        12:30 AM
      </td>
      <td>
        30 mins
      </td>
      <td>
        <div>
          Getting Started With Computer Vision Using Clarifai
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        Walter Light Hall - Room 210
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Hardware Opens 2019-02-02T01:00:00"
    >
      <td>
        1:00 AM
      </td>
      <td>
        1 hr
      </td>
      <td>
        <div>
          Hardware Opens
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        ILC Design Bay
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Slideshow Karaoke 2019-02-02T01:00:00"
    >
      <td>
        1:00 AM
      </td>
      <td>
        1 hr
      </td>
      <td>
        <div>
          Slideshow Karaoke
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        Walter Light Hall - Room 210
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Snack (Bubble Tea) 2019-02-03T00:00:00"
    >
      <td>
        12:00 AM
      </td>
      <td>
        1 hr
      </td>
      <td>
        <div>
          Snack (Bubble Tea)
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        Walter Light Hall Auditorium
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Escape Room 2019-02-03T01:00:00"
    >
      <td>
        1:00 AM
      </td>
      <td>
        1 hr
      </td>
      <td>
        <div>
          Escape Room
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        ILC - Room 119
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Breakfast 2019-02-03T07:30:00"
    >
      <td>
        7:30 AM
      </td>
      <td>
        1 hr
      </td>
      <td>
        <div>
          Breakfast
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        ILC - Room 109
      </td>
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Hacking Ends 2019-02-03T09:00:00"
    >
      <td>
        9:00 AM
      </td>
      <td>
        1 hr
      </td>
      <td>
        <div>
          Hacking Ends
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td />
    </tr>
    <tr
      css={
        Object {
          "backgroundColor": "#F6F6F6",
        }
      }
      key="Closing Ceremonies 2019-02-03T13:00:00"
    >
      <td>
        1:00 PM
      </td>
      <td>
        2 hrs
      </td>
      <td>
        <div>
          Closing Ceremonies
        </div>
        <div
          css={
            Object {
              "color": "#4d4d4d",
              "fontSize": "13px",
              "marginTop": "5px",
            }
          }
        />
      </td>
      <td>
        Kingston Hall
      </td>
    </tr>
  </tbody>
</table>
`;
